src/libraries/Common/src/Interop/Android/System.Security.Cryptography.Native.Android/Interop.EVP.Cipher.cs

```diff
@@ -0,0 +1,235 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Runtime.InteropServices;
+using Microsoft.Win32.SafeHandles;
+
+internal static partial class Interop
+{
+    internal static partial class Crypto
+    {
+        [DllImport(Libraries.CryptoNative, EntryPoint = "CryptoNative_EvpCipherCreate2")]
+        internal static extern SafeEvpCipherCtxHandle EvpCipherCreate(
```

Comment:
Is there too much code sharing to get rid of the OpenSsl-based type names here? EVP_CIPHER and EVP_CIPHER_CTX are OpenSSL-isms.

Reply:
Yeah we're currently reusing the handle types and the matching DllImports for the alloc/release methods. I can separate them and do all the renaming. It'll just make this PR a lot larger.

Reply:
I've taken more of a look and we're still sharing too much code to rename all of these. We're still sharing all of the non-GCM/CCM AES implementation, which uses all of these functions.